wagtail/core/templatetags/wagtailcore_tags.py

```diff
@@ -34,8 +34,26 @@ def pageurl(context, page):
 
 @register.simple_tag(takes_context=True)
 def slugurl(context, slug):
-    """Returns the URL for the page that has the given slug."""
-    page = Page.objects.filter(slug=slug).first()
+    """
+    Returns the URL for the page that has the given slug.
+
+    First tries to find a page on the current site. If that fails or a request
+    is not available in the context, then returns the URL for the first page
+    that matches the slug on any site.
+    """
+
+    try:
+        current_site = context['request'].site
+        page = Page.objects.filter(
```

Comment:
Please change this to: ``Page.objects.in-site(current_site).filter(slug=slug).first()``